.github/workflows/integration-tests.yml

```diff
@@ -173,6 +173,50 @@ jobs:
           python3 -m pytest -vs . --reruns 10
           sudo nvidia-smi -i 0 -rgc
 
+  Integration-Tests-Shared-Middle-Layer:
```

Comment:
could we have it be a different workflow file that triggers on every `main` commit instead of every PR? I'd like to do the same thing for 3P backends too. Seems to make more sense to only run required workflows on every PR and be more lenient otherwise

Reply:
I'm actually happy to merge this PR and do this if that's ok with you!